tree/ntuplebrowse/v7/src/RNTupleBrowser.cxx

```diff
@@ -0,0 +1,157 @@
+{
+   for (auto ptr : fNTupleBrowsePtrVec) {
+      delete ptr;
+   }
+   // No need to delete TH1F* fCurrentTH1F, because it's automatically deallocated when TBrowser is closed.
+}
+
+void ROOT::Experimental::RNTupleBrowser::Browse(TBrowser *b)
+{
+   ROOT::Experimental::RBrowseVisitor browseVisitor(b, this);
+   fReaderPtr->GetModel()->GetRootField()->TraverseVisitor(browseVisitor);
+}
+   
+//---------------------------- NTupleFieldElementFolder ---------------------------
+   
+ClassImp(ROOT::Experimental::RNTupleFieldElementFolder);
+
+void ROOT::Experimental::RNTupleFieldElementFolder::Browse(TBrowser *b)
+{
+   RBrowseVisitor browseVisitor(b, fRNTupleBrowserPtr);
+   if (fFieldPtr) {
+      fFieldPtr->TraverseVisitor(browseVisitor);
+   } else {
+      fNtupleReaderPtr->GetModel()->GetRootField()->TraverseVisitor(browseVisitor);
+   }
+}
+
+void ROOT::Experimental::RNTupleFieldElementFolder::AddBrowse(TBrowser *b)
+{
+   // in case of unit test, b->Add(this) will call the Add function implemented in TBrowser.h, despite the pointer not pointing to an instance from TBrowser.h (instead it points to the one created in ntuple_browse.cxx) For that reason, b->Add(this) is not called, when (fUnitTest != 0)<=> program is executed in an unittest.
+   if (fRNTupleBrowserPtr->GetfUnitTest()) {
+      fRNTupleBrowserPtr->IncreasefUnitTest();
+   } else {
+      b->Add(this);
+   }
+}
+   
+//---------------------------- NTupleFieldElement --------------------
+   
+ClassImp(ROOT::Experimental::RNTupleFieldElement);
+
+void ROOT::Experimental::RNTupleFieldElement::AddBrowse(TBrowser *b)
+{
+   if (fRNTupleBrowserPtr->GetfUnitTest()) {
+      fRNTupleBrowserPtr->IncreasefUnitTest();
+   } else {
+      b->Add(this);
+   }
+}
+
+void ROOT::Experimental::RNTupleFieldElement::Browse(TBrowser* /*b*/)
+{
+   switch(fType) {
+      case fieldDatatype_nonNumeric:
+      case fieldDatatype_notkLeaf:
+      case fieldDatatype_parentIsVec:
+      case fieldDatatype_noHist: break;
+      case fieldDatatype_float: TemplatedBrowse<float>(false); break;
+      case fieldDatatype_double: TemplatedBrowse<double>(false); break;
+      case fieldDatatype_Int32: TemplatedBrowse<std::int32_t>(true); break;
+      case fieldDatatype_UInt32: TemplatedBrowse<std::uint32_t>(true); break;
+      case fieldDatatype_UInt64: TemplatedBrowse<std::uint64_t>(true); break;
+   }
+}
+
+template <typename T>
+void ROOT::Experimental::RNTupleFieldElement::TemplatedBrowse(bool isIntegraltype)
+{
+   const char* name = (const char*)fName;
+   auto ntupleView = fReaderPtr->GetView<T>(std::string(name));
+   auto numEntries{ fReaderPtr->GetNEntries() };
+   if (numEntries == 0) return; // no histogram
+   
+   double max{static_cast<double>(ntupleView(0))}, min{static_cast<double>(ntupleView(0))};
+   for(unsigned long long i = 1; i < numEntries; ++i) {
+      if (ntupleView(i) > max) {
+         max = ntupleView(i);
+      } else if (ntupleView(i) < min) {
```

Comment:
I guess min will often remain zero (not be the minimum of the set).  Initialize min and max using `numeric_limits` from the limits header.  The code to update `min` and `max` can also use `std::min()` and `std::max()`.